org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/quickfix/ConstructorsQuickFixTest.xtend

```diff
@@ -379,4 +379,403 @@ class ConstructorsQuickFixTest extends AbstractWollokQuickFixTestCase {
 		assertQuickfix(initial, result, Messages.WollokDslQuickFixProvider_remove_constructor_name, 2)
 	}
 
+	@Test
+	def addNewConstructorInRefClassNoParams(){
+		val initial = #[
+		'''
+		class A {
+			const y
+			constructor(_y){
+				y = _y
+			}
+		
+			method someMethod(){
+				return null
+			}
+		}
+		
+		class B {
+			method b() {
+				const a = new A()
+				console.println(a.toString())
+			}
+		}
+		''']
+
+		val result = #[
+		'''
+		class A {
+			const y
+			constructor(_y){
+				y = _y
+			}
+			constructor() {
+				//TODO: Autogenerated Code ! 
+			}
+		
+			method someMethod(){
+				return null
+			}
+		}
+		
+		class B {
+			method b() {
+				const a = new A()
+				console.println(a.toString())
+			}
+		}
+		''']
+		assertQuickfix(initial, result, Messages.WollokDslQuickFixProvider_create_constructor_class_name)
+	}
+	
+	@Test
+	def addNewConstructorInRefClassOneParam(){
+		val initial = #[
+		'''
+		class A {
+			var y = 0
+			method someMethod(){
+				return y
+			}
+		}
+		
+		class B {
+			method b() {
+				const a = new A("hola")
+				console.println(a.toString())
+			}
+		}
+		''']
+
+		val result = #[
+		'''
+		class A {
+			var y = 0
+			constructor(param1) {
+				//TODO: Autogenerated Code ! 
+			}
+			method someMethod(){
+				return y
+			}
+		}
+		
+		class B {
+			method b() {
+				const a = new A("hola")
+				console.println(a.toString())
+			}
+		}
+		''']
+		assertQuickfix(initial, result, Messages.WollokDslQuickFixProvider_create_constructor_class_name)
+	}
+	
+	@Test
+	def addNewConstructorInRefClassSeveralParams(){
+		val initial = #[
+		'''
+		class A {
+			var y = 0
+			method someMethod(){
+				return y
+			}
+		}
+		
+		class B {
+			method b() {
+				const a = new A("hola", 1)
+				console.println(a.toString())
+			}
+		}
+		''']
+
+		val result = #[
+		'''
+		class A {
+			var y = 0
+			constructor(param1, param2) {
```

Comment:
It would be nice to create names such as "aString", "aNumber", 'aGolonrina".
But maybe in another ticket.

Reply:
We both know the answer :smile: